Add embed mode options in the Share UI

ajwild opened the pull request:
## Summary

This addresses a few of the capabilities listed in #9575 - specifically showing the query input, date picker and filters. I haven't looked into the zooming and panel moving/resizing options yet.

I also made it possible to hide the top nav menu items (Full Screen, Share, Clone, Edit) but this exposed a bug so it may be simpler to remove this functionality if it's not really required.

I'll add a few comments below where some feedback would be appreciated.

![Embedded snapshot](https://user-images.githubusercontent.com/5101376/75227122-f3aa1900-57f0-11ea-9155-e31b8334c77e.png)

![Embedded saved object](https://user-images.githubusercontent.com/5101376/75227211-1dfbd680-57f1-11ea-8e3a-392e0b8d3077.png)

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server)
- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

kibanamachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

timroes commented:
Jenkins, test this

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

ajwild commented on src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx:
```diff
@@ -713,8 +715,44 @@ export class DashboardAppController {
+
+    $scope.showQueryInput = () => $scope.isVisible || displayIfUrlParam('show-query-input');
+
+    $scope.showDatePicker = () => $scope.isVisible || displayIfUrlParam('show-date-picker');
```
I've used functions for all of these, similar to `showFilterBar`, because this made it easier to test while developing as I could update the URL and see the changes immediately. My thoughts are that this is better than setting up watchers similar to `showSaveQuery`, and also better than only checking the value on page load in case users want to enable certain functionality dynamically.

majagrubic replied:
It's a bit repetitive, but I'm ok with this as this code will change (somewhat) soon 😬

ajwild commented on src/plugins/dashboard_embeddable_container/public/plugin.tsx:
```diff
@@ -70,6 +70,7 @@ export class DashboardEmbeddableContainerPublicPlugin
 
     const ExitFullScreenButton: React.FC<ExitFullScreenButtonProps> = props => {
+      // TODO: Only call useHideChrome if chrome is visible
       useHideChrome();
```
This line is the cause of the main issue remaining with the "show top nav menu" option. When called it hides the chrome (and I've made my changes react to it as well) which is fine as it should already be hidden in embedded mode, but when exiting full screen mode it calls the clean up callback which always displays the chrome. Ideally this would revert back to the previous state, or just skip the whole function if `isVisible$` is false, but I wasn't sure the best way of doing this with RxJS. It feels like a hack if I have to subscribe to get the current value and then unsubscribe immediately afterwards. I'm not very familiar with RxJS, so what would be the correct "reactive" way of doing this?

ajwild commented on src/plugins/data/public/ui/query_string_input/query_bar_top_row.tsx:
```diff
@@ -348,7 +348,7 @@ function QueryBarTopRowUI(props: Props) {
       responsive={!!props.showDatePicker}
       gutterSize="s"
-      justifyContent="flexEnd"
+      justifyContent="flexStart"
```
I've made this change because the date picker looks out of place when it's right aligned and the query input is not displayed - mostly because the filter bar is left aligned below it. Is there any harm in doing this?

majagrubic replied:
I don't see any harm in doing this.

ajwild replied:
I generally prefer the right alignment, and I feel keeping the date picker location consistent is a good thing. However, in embedded mode this may not be as important because the end user may not be a frequent Kibana user, and I think that it looks out of place when there is content on the lines above and below like this:
![flexEnd](https://user-images.githubusercontent.com/5101376/75495885-fd0acf80-5a02-11ea-97a0-beea66a2e896.png)

I can make this a prop so that it doesn't impact the other views, but it would be good to hear some more thoughts. I also have no strong opinion on this, but I made the change to find out if it's preferred. I'd be happy to change it back.

cchaos replied:
I agree with you @ajwild. We are aware that this is not ideal and are continuing a search for a better solution. The better solution though involves lots of folks across different teams coming to a conclusion so it's taking a while to nail down. Consistency is the main factor here, and we should try to keep the global time filters in the same place across all apps (even when no query bar exists).

In embed mode 🤷‍♀ , it's probably okay to shift.

flash1293 replied:
@ajwild I like the prop solution in that case

lizozom replied:
I think it's okay to do a shift, as long as it's not complicating the API \ code.
As the design of this will soon change anyway, I think we could go with an 80:20 solution, allowing having the datepicker in embedded mode, even if it's not perfectly positioned ATM.

ajwild commented on src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx:
```diff
@@ -54,6 +60,21 @@ export function TopNavMenu(props: TopNavMenuProps) {
   }
 
+  function renderMenu() {
+    if (!showTopNavMenu) return;
```
The new `showTopNavMenu` prop avoids rendering the `kbnTopNavMenu` element when it's not required, and allows hiding the menu even if menu items are passed. This could also be done by conditionally deciding whether to pass no `config`, however it would still render the empty element before this change and I didn't want to affect the existing behaviour in case something else is relying on the additional padding the element introduces.

majagrubic replied:
I think it's ok to remove it.

ajwild commented on src/plugins/share/public/components/url_panel_content.tsx:
```diff
@@ -186,13 +203,25 @@ export class UrlPanelContent extends Component<Props, State> {
     }
-    return `${url}${embedQueryParam}`;
+    return `${url}${embedQueryParams}`;
   };
```
Once #54403 is merged this would need implementing in src/plugins/share/server/routes/goto.ts  too.

flash1293 replied:
This is now solved differently, current solution should be fine.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/)
* Commit: bbca5a761e6da5a246f223457264c29e096213e3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/","number":"28714","commit":"bbca5a761e6da5a246f223457264c29e096213e3"}]}
PIPELINE-->

majagrubic commented on src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx:
```diff
@@ -713,8 +715,44 @@ export class DashboardAppController {
     }
 
+    const hasUrlParam = (param: string): boolean =>
+      param in parse(location.hash.slice(1), true).query;
```
We already have a mechanism for checking URL parameters. Can you make use of `$routeParams` here? Eg:
https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx#L354

majagrubic commented on src/plugins/share/public/components/url_panel_content.tsx:
```diff
@@ -60,11 +60,20 @@ enum ExportUrlAsType {
 interface State {
   exportUrlAs: ExportUrlAsType;
   useShortUrl: boolean;
+  showTopNavMenu: boolean;
```
Since this component is to be reused across Kibana, I'm not a big fan of bloating it with app-specific props. Can we make a HOC component around `UrlPanelContent`?

majagrubic commented on src/plugins/share/public/components/url_panel_content.tsx:
```diff
@@ -77,6 +86,10 @@ export class UrlPanelContent extends Component<Props, State> {
     this.state = {
       exportUrlAs: ExportUrlAsType.EXPORT_URL_AS_SNAPSHOT,
       useShortUrl: false,
+      showTopNavMenu: false,
```
Since this component is reused across different apps, these will be applied everywhere. Can we make a HOC component around `UrlPanelContent` that will be dashboard specific?

flash1293 replied:
I agree with @majagrubic here - the `share` plugin should be generic and be usable in all applications, so application-specific logic should be part of the application.

I propose the following API:
Currently `toggleShareContextMenu` is called in https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx#L858 (eventually defined in https://github.com/elastic/kibana/blob/master/src/plugins/share/public/services/share_menu_manager.tsx#L59 )

Let's add a new optional property to `ShowShareMenuOptions` called
```ts
// Props are all props of the UrlPanelContent component
urlParamExtensions: Array<{
  paramName: string;
  component: ComponentType<Props & { setParamValue: (value: string) => void; }>
}>
```

This should allow us to move all of the dashboard-specific stuff into the dashboard controller and keep the generic logic of rendering a list of components, keeping state for a list of params and appending them to the url here in the share plugin.

lizozom replied:
IMO These options are as relevant to Visualize as well, don't you think? @flash1293

flash1293 replied:
I thought the plan for this would be it becoming a single integration point for sharing, also usable from apps like Canvas, Graph and so on. But maybe that's not necessary, we can also put it behind a prop which defaults to `false` directly in the share code and only disable it for dashboards for now. No strong opinion on this, whatever you are more comfortable with :)

majagrubic commented on src/plugins/share/public/components/url_panel_content.tsx:
```diff
@@ -186,13 +203,25 @@ export class UrlPanelContent extends Component<Props, State> {
+      ['&show-top-nav-menu=true', this.state.showTopNavMenu],
+      ['&show-query-input=true', this.state.showQueryInput],
+      ['&show-date-picker=true', this.state.showDatePicker],
+      ['&hide-filter-bar=true', !this.state.showFilterBar], // Inverted to keep default behaviour for old links
```
Can we name all parameters `hide-something` for consistency?

lizozom replied:
I'd prefer show, just so we're consistent with our own flags.

ajwild replied:
If I change it to use `&show-filter-bar=true`, then all previously generated links (which users may have already embedded into websites etc.) will begin hiding the filter bar because this value won't be set.

lizozom replied:
Well, then lets go for `hide-something` like @majagrubic suggested?

majagrubic commented on src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx:
```diff
@@ -41,8 +42,13 @@ export type TopNavMenuProps = StatefulSearchBarProps & {
+export function TopNavMenu(props: TopNavMenuProps): ReactElement | null {
+  const { config, showTopNavMenu, showSearchBar, ...searchBarProps } = props;
+
+  if (!showTopNavMenu && (!showSearchBar || !props.data)) {
```
Ideally, we wouldn't even be rendering this component is `showTopNavMenu` is false.

ajwild replied:
I could rename it to `showMenu`, as I think it's still useful to pick if the menu and search bar are rendered separately. However, I'll take another look at using `config` instead, and moving the conditional to the dashboard when getting the menu items to show.

gchaps commented:
How about adding a label over the "Show" toggle buttons to reduce the amount of text the user needs to read.  Here is my suggested wording:

**Include**

Top nav
Query
Time filter
Filter bar

Also, for these four items, I suggest removing the tooltip.

cchaos commented:
I agree with @gchaps. And at that point, the group should just be a `EuiCheckboxGroup` (for spacing and accessibility concerns) which you will be able to find an example of on this page: https://elastic.github.io/eui/#/forms/form-controls and use an `EuiHorizontalRule` to separate the concerns.

<img width="322" alt="Screen Shot 2020-02-27 at 16 17 11 PM" src="https://user-images.githubusercontent.com/549577/75487624-ba061900-597c-11ea-99e7-c24b5b4eca39.png">

Let me know if you'd like help implementing these UI controls.

ajwild commented:
Thanks for all the feedback! I hope to make the changes requested on Monday, and I'll reach out if any questions come up.

lizozom commented:
@elasticmachine merge upstream

lizozom commented on src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app.html:
```diff
@@ -4,14 +4,19 @@
+    show-date-picker="showDatePicker()"
+    show-filter-bar="showFilterBar()"
 
+    justify-content="justifyContent"
```
I don't think we should be passing individual styling options here.
If absolutely necessary, we could pass `classNames`
Even better would be to let `QueryBarTopRow` decide for itself, according to which components are enabled.

lizozom commented on src/plugins/dashboard_embeddable_container/public/embeddable/dashboard_container.tsx:
```diff
@@ -53,6 +53,7 @@ export interface DashboardContainerInput extends ContainerInput {
   useMargins: boolean;
   title: string;
   description?: string;
+  isEmbedded: boolean;
```
An embeddable is always embedded within some other container.
In the context of this PR, what does `isEmbedded` mean?
We might need a different terminiology.

lizozom commented on src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx:
```diff
@@ -270,6 +271,7 @@ export class DashboardAppController {
         viewMode: dashboardStateManager.getViewMode(),
         panels: embeddablesMap,
         isFullScreenMode: dashboardStateManager.getFullScreenMode(),
+        isEmbedded: Boolean($routeParams.embed),
```
@majagrubic would it be correct to call `isEmbedded` :arrow_right: `isShared` \ `isExternal` \ `isEmbeddedExternally`?

majagrubic replied:
`isEmbeddedExternally` sounds good.

flash1293 commented:
Hi @ajwild , sorry for the delay in engagement here.
If you are still willing to get this PR forward I'm more than happy to work with you on it - from now on I'll be much faster in responding.

I tried to update your PR to current master, but I'm not sure whether I resolved the conflicts correctly as it seems like there are some bugs (e.g. couldn't get the timepicker to show): 
My attempt on updating can be found here: https://github.com/flash1293/kibana/tree/embed-mode-options

Could you take a look?

flash1293 commented on src/plugins/share/public/components/url_panel_content.tsx:
```diff
@@ -394,4 +434,37 @@ export class UrlPanelContent extends Component<Props, State> {
+    }
+
+    const checkboxes = [
+      ['topMenu', 'Top menu'],
```
These should be internationalized with `i18n.translate` like here: https://github.com/elastic/kibana/pull/58435/files#diff-2fa41be73e54b8cb0480dfc410ef33dcR301

flash1293 commented on src/plugins/share/public/components/url_panel_content.tsx:
```diff
@@ -100,8 +117,8 @@ export class UrlPanelContent extends Component<Props, State> {
           {this.renderExportAsRadioGroup()}
-
           {this.renderShortUrlSwitch()}
+          {this.renderUrlParamExtensions()}
```
This is always rendered even when the embedding application doesn't act on the parameters - e.g. Visualize

If we are not moving this completely into the dashboard plugin, we should at least put it behind a config flag.

flash1293 commented on src/plugins/share/public/components/url_panel_content.tsx:
```diff
@@ -190,13 +207,25 @@ export class UrlPanelContent extends Component<Props, State> {
-    const embedQueryParam = '?embed=true';
+  private getEmbedQueryParams = (): string => {
+    return [
+      ['&show-top-menu=true', this.state.urlParamsSelectedMap.topMenu],
```
These magic strings are used here and in the dashboard plugin - to make sure they stay the same, please put them in constants, export them from the root level `index.ts` file of the share plugin and import them in the dashboard plugin.

flash1293 commented on src/plugins/share/public/components/url_panel_content.tsx:
```diff
@@ -282,6 +311,17 @@ export class UrlPanelContent extends Component<Props, State> {
 
+  private handleUrlParamChange = (optionId: string): void => {
+    const param = optionId as keyof UrlParams;
+    const stateUpdate: Partial<State> = {
```
nit: If you don't declare this to be a `Partial<State>`, then you don't need the type cast in line 322.

ajwild commented:
@flash1293 I just started looking at this again today, and I'm currently working on using `urlParamExtensions`. I've also rebased to master.

ajwild commented:
@flash1293 I've made a few changes in the latest commit which could do with a thorough review. I just noticed that it's no longer possible to embed when "Saved object" is selected within "Embed code" share panel - it would be good to know if this is deliberate.

The remaining things I can see:
1. Remove `justify-content="justifyContent"` and automatically calculate position, or use CSS classes instead?
2. Decide if a "breaking change" should be introduced by changing `hide-filter-bar=true` to `show-filter-bar=true`
3. Decide how to handle Short URLs when toggling the URL param extensions

Regarding point 3, once a short URL is generated it is cached, and toggling the params does not invalidate this so they are not applied. One option to fix this would be to keep generating a new short URL each time a param is toggled. Alternatively, the short URL generation could be postponed, possibly by having the button at the bottom of the panel as "Generate short URL" and when complete show the "Copy iFrame code" button.

flash1293 commented:
@ajwild Thanks for your work.

> I just noticed that it's no longer possible to embed when "Saved object" is selected within "Embed code" share panel - it would be good to know if this is deliberate.

No, that's definitely a bug, thanks for that. I'll create a separate issue for this.

> Remove justify-content="justifyContent" and automatically calculate position, or use CSS classes instead?

I'm fine with the `justifyContent` setting, makes the component more flexible and might come in handy at a later point. @lizozom what do you think?

> Decide if a "breaking change" should be introduced by changing `hide-filter-bar=true` to `show-filter-bar=true`

I think it's not worth it (especially in a minor version) - this would probably break a ton of integrations hardcoded in places like wiki pages and so on.

> Decide how to handle Short URLs when toggling the URL param extensions

Can we generate the short url when the "Copy iFrame code" button is clicked?

FYI - there is currently a PR in flight moving the top nav out of the angular template: https://github.com/elastic/kibana/pull/61611 It will conflict with this PR, but should be easy to resolve - basically the props definition gets moved into the controller.

ajwild commented:
> Can we generate the short url when the "Copy iFrame code" button is clicked?

It's possible, but there are browser enforced time limits between clicking a button and adding text to clipboard. On the current versions of Firefox and Chromium this is 5 seconds, so this should usually be enough time. However, if the URL generation takes longer than 5 seconds for whatever reason then copy to clipboard won't work on the first click. If this is an acceptable situation/risk then I'll start work on the single button, otherwise the two button approach will be safer.

flash1293 commented:
Ah, good point.

What about moving the short url switch below the options like this

<img width="275" alt="Screenshot 2020-04-08 at 12 49 04" src="https://user-images.githubusercontent.com/1508364/78776232-ccba3780-7997-11ea-973b-6d992e11c54f.png">

And then un-toggling the switch every time a user manipulates the options? That's basically the same behavior like a separate "generate short url" button but stays closer to the current layout and as the short url switch is the last option will hopefully not be noticed by a lot of users because they specify their includes before switching short url.

@cchaos what do you think about this solution? The context is that we can't generate the short URL for technical reasons when the "Copy iframe code" button is pressed, but changing the options requires us to re-generate.

ajwild commented:
I like that solution, but I'll wait for more feedback before doing anything.

cchaos commented:
Hmm, unselecting (some) of the user's options as they toggle others, especially in such a small amount, is a bit of an anti-pattern. If I'm gathering this correctly. it's all about a timing issue? Is it possible to put the Copy button into the `isLoading` state, while the URL is being generated? Then only once the URL is ready, will the button be re-enabled.

flash1293 commented:
@cchaos It's not just a timing issue, the URL would have to be re-generated every time a checkbox is checked/unchecked, which involves creating a saved object on the server. The solution with unchecking short url automatically is avoiding to create lots of orphaned saved objects while the user is still configuring.

flash1293 commented:
@ajwild I discussed this offline with @cchaos and we came to the conclusion that the best tradeoff here is to keep the "short url" switch toggled when the user changes the include options and re-generate a new short url on the fly, but do the reordering of the UI I mentioned above putting the short url switch right above the button.

Due to the re-ordering we minimize the number of cases unnecessary short urls are generated, but we still avoid frustrating the user by reverting a setting they made - as it won't happen super often, the number of stale short url objects should be manageable.

In summary: Let's go with your first suggestion (https://github.com/elastic/kibana/pull/58435#issuecomment-610007309) "keep generating a new short URL each time a param is toggled", but put the toggle in the bottom of the popover.

Thanks again for working at this, it's much appreciated!

ajwild commented:
@flash1293 sorry for the delay on this, there were a lot of changes during the rebase so it took me a while to get around to working through them all. I refactored a few things during the rebase so it'll probably need a full review again. The main change was removing the `$scope.show*` functions and replacing them with the `const forceShow*` logic. This cleaned up the code quite a bit, but it means that changes to the URL query params are no longer reflected in the UI straight away and a page load/refresh is required. Let me know if you want me to add something to watch the URL params for changes, otherwise I think I've addressed everything in the comments now.

flash1293 commented:
@elasticmachine merge upstream

flash1293 commented:
Jenkins, test this.

flash1293 commented:
I tested in Chrome and it seems like everything is working as expected, great job @ajwild !
@majagrubic @cchaos @lizozom could you have another look?

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/)
* Commit: 07aa92deb835ea89c4063910e4be508e1252be20
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/execution/node/237/log)

### History
* :green_heart: [Build #28714](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/) succeeded bbca5a761e6da5a246f223457264c29e096213e3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/","number":"28714","commit":"bbca5a761e6da5a246f223457264c29e096213e3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/","number":"47914","commit":"07aa92deb835ea89c4063910e4be508e1252be20"}]}
PIPELINE-->

flash1293 commented:
@ajwild To make the build pass you have to run the doc generation script and check in the result as described here: https://github.com/elastic/kibana/pull/58435/checks?check_run_id=674432964

cchaos commented:
@ajwild, I noticed a couple things and though it'd just be easier to pass you a PR, so here it is: https://github.com/ajwild/kibana/pull/1 Other than that it 👍 LGTM!

majagrubic commented on src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx:
```diff
@@ -59,6 +68,22 @@ export function TopNavMenu(props: TopNavMenuProps) {
   }
 
+  function renderMenu(className: string) {
+    if (!config || config.length === 0) return;
```
Nit: let's add `return null` here.

majagrubic commented on src/plugins/share/public/components/url_panel_content.tsx:
```diff
@@ -180,28 +190,51 @@ export class UrlPanelContent extends Component<Props, State> {
+          !this.state.urlParams[key] ||
+          !this.state.urlParams[key][queryParam]
+        ) {
+          return queryAccumulator;
```
nit: could we use lodash `_.get` here to clear up this syntax a bit?

flash1293 commented on src/plugins/kibana_react/public/exit_full_screen_button/exit_full_screen_button.tsx:
```diff
@@ -24,6 +24,7 @@ import { EuiIcon, EuiFlexGroup, EuiFlexItem, EuiText } from '@elastic/eui';
 
 export interface ExitFullScreenButtonProps {
   onExitFullScreenMode: () => void;
+  toggleChrome?: boolean;
```
Just noticed another thing, @ajwild - the `toggleChrome` flag is only necessary in the dashboard integration, we shouldn't make it part of the `ExitFullScreenButton` interface. Please but this into the dashboard plugin: https://github.com/elastic/kibana/pull/58435/files#diff-42d3b4a54aaaa4907d311e7020d85682R193

```
const ExitFullScreenButton: React.FC<ExitFullScreenButtonProps & { toggleChrome: boolean; }> = props => {
```

lizozom commented:
@elasticmachine merge upstream

ajwild commented on src/plugins/data/public/ui/search_bar/create_search_bar.tsx:
```diff
@@ -193,6 +193,7 @@ export function createSearchBar({ core, storage, data }: StatefulSearchBarDeps)
           showQueryBar={props.showQueryBar}
           showQueryInput={props.showQueryInput}
           showSaveQuery={props.showSaveQuery}
+          justifyContent={props.justifyContent}
```
Sure, I'll do that today (along with the other changes requested above). Should I remove commit 7814760bef3b67749bbd5963f920356e58b48904 or just revert it? I'm not sure how it gets picked up in the release process.

ajwild commented:
Just the functional tests left. It's taking a while to get them running, so I'll have to come back to them another time.

flash1293 commented:
Jenkins, test this.

flash1293 commented:
@ajwild I'm not 100% sure what's causing this - I kicked off a CI build, let's see how things look there.

For reference, I always start the functional tests like this:
```sh
node scripts/functional_tests_server.js # wait till server is up and compilation is finished

# in another tab
node scripts/functional_test_runner.js --grep="dashboard" # might be acommpanied by .only flags for specific test suites
```

ajwild commented:
Thanks for the help @flash1293. I had been trying to get `yarn test:ftr` to work, but the separate commands helped. I found the [Functional Testing guide](https://www.elastic.co/guide/en/kibana/current/development-functional-tests.html) and switched to the Firefox config as Chromium wasn't working for me either (missing manifest file). Now it's running so I should be able to extend the tests today.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/)
* Commit: 0d92ea5191f7f50e763f12c48c9a763ff6b6debc
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/execution/node/492/log)
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/execution/node/259/log)

### History
* :broken_heart: [Build #47914](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/) failed 07aa92deb835ea89c4063910e4be508e1252be20
* :green_heart: [Build #28714](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/) succeeded bbca5a761e6da5a246f223457264c29e096213e3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/","number":"28714","commit":"bbca5a761e6da5a246f223457264c29e096213e3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/","number":"47914","commit":"07aa92deb835ea89c4063910e4be508e1252be20"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/","number":"49076","commit":"0d92ea5191f7f50e763f12c48c9a763ff6b6debc"}]}
PIPELINE-->

flash1293 commented:
@ajwild Existing tests look fine, the remaining i18n error should be super simple to fix (https://github.com/elastic/kibana/pull/58435/checks?check_run_id=692113200) - you can just introduce a new i18n id within the dashboard namespace for this.

ajwild commented:
Hopefully the two latest commits fix things. Is it OK that I piggy-backed on to existing `data-test-subject` values? Would it be better to introduce new values, or created `exists()` functions within the test services?

flash1293 commented on test/functional/apps/dashboard/embed_mode.js:
```diff
@@ -54,9 +62,38 @@ export default function({ getService, getPageObjects }) {
     });
 
+    it('shows or hides elements based on URL params', async () => {
+      const topMenuShown = await testSubjects.exists('top-nav');
```
This can be shortened to `await testSubjects.existsOrFail('top-nav');`

flash1293 commented on test/functional/apps/dashboard/embed_mode.js:
```diff
@@ -54,9 +62,38 @@ export default function({ getService, getPageObjects }) {
+      await browser.get(newUrl.toString(), useTimeStamp);
+
+      await retry.try(async () => {
+        const topMenuShown = await testSubjects.exists('top-nav');
```
This can be shortened to `await testSubjects.existsOrFail('top-nav');`

flash1293 commented:
Re-using existing `data-test-subjects` is the right approach - I think the new test is fine, just left a small nit about using another helper.

ajwild commented:
@flash1293 I've added the helper functions to the test. Much cleaner now :+1:

flash1293 commented:
Jenkins, test this.

flash1293 commented on src/plugins/dashboard/public/application/dashboard_app_controller.tsx:
```diff
@@ -977,7 +977,7 @@ export class DashboardAppController {
           ].map(([urlParam, translationId, defaultMessage]) => ({
             id: urlParam,
-            label: i18n.translate(`share.urlPanel.${translationId}`, {
+            label: i18n.translate(`dashboard.embedUrlParamExtension.${translationId}`, {
```
Missed that earlier - to make it possible to statically collect all existing i18n ids, `translate` is only allowed to be called with a constant. It's a bit redundant sometimes but has a lot of benefits for the downstream process.

In that case I suggest:
```ts
const checkboxes = [
            [UrlParams.SHOW_TOP_MENU, i18n.translate('dashboard.embedUrlParamExtension.topMenu', { defaultMessage: 'Top menu' })],
            [UrlParams.SHOW_QUERY_INPUT, i18n.translate('dashboard.embedUrlParamExtension.query', { defaultMessage: 'Query'})],
            [UrlParams.SHOW_TIME_FILTER, i18n.translate('dashboard.embedUrlParamExtension.timeFilter', { defaultMessage: 'Time filter'})],
            [UrlParams.SHOW_FILTER_BAR, i18n.translate('dashboard.embedUrlParamExtension.filterBar', { defaultMessage: 'Filter bar'})],
].map(([urlParam, label]) => ({
            id: urlParam,
            label
```

ajwild replied:
I've gone with a simple array of objects (without the map). Once Prettier had formatted it they took up the same number of lines, so this seems clearer even if it means the prop names are repeated.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/)
* Commit: 6e55e8189b0dc4acd913e4b2b619729747ecb42f
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/execution/node/255/log)

### History
* :broken_heart: [Build #49076](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/) failed 0d92ea5191f7f50e763f12c48c9a763ff6b6debc
* :broken_heart: [Build #47914](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/) failed 07aa92deb835ea89c4063910e4be508e1252be20
* :green_heart: [Build #28714](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/) succeeded bbca5a761e6da5a246f223457264c29e096213e3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/","number":"28714","commit":"bbca5a761e6da5a246f223457264c29e096213e3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/","number":"47914","commit":"07aa92deb835ea89c4063910e4be508e1252be20"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/","number":"49076","commit":"0d92ea5191f7f50e763f12c48c9a763ff6b6debc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/","number":"49123","commit":"6e55e8189b0dc4acd913e4b2b619729747ecb42f"}]}
PIPELINE-->

flash1293 commented:
Jenkins, test this.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49149/)
* Commit: 9fb085033c333f670ba06b088a9690a05587d026

### History
* :broken_heart: [Build #49123](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/) failed 6e55e8189b0dc4acd913e4b2b619729747ecb42f
* :broken_heart: [Build #49076](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/) failed 0d92ea5191f7f50e763f12c48c9a763ff6b6debc
* :broken_heart: [Build #47914](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/) failed 07aa92deb835ea89c4063910e4be508e1252be20
* :green_heart: [Build #28714](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/) succeeded bbca5a761e6da5a246f223457264c29e096213e3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/","number":"28714","commit":"bbca5a761e6da5a246f223457264c29e096213e3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/","number":"47914","commit":"07aa92deb835ea89c4063910e4be508e1252be20"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/","number":"49076","commit":"0d92ea5191f7f50e763f12c48c9a763ff6b6debc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/","number":"49123","commit":"6e55e8189b0dc4acd913e4b2b619729747ecb42f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49149/","number":"49149","commit":"9fb085033c333f670ba06b088a9690a05587d026"}]}
PIPELINE-->

flash1293 commented:
@majagrubic @lizozom If you have the time another look would be appreciated :)

majagrubic commented on test/functional/apps/dashboard/embed_mode.js:
```diff
@@ -54,9 +62,28 @@ export default function({ getService, getPageObjects }) {
       });
     });
 
+    it('shows or hides elements based on URL params', async () => {
```
thanks for adding this!

thabanicus commented:
Hi Guys. 

I'm cloned the project and setup the dev environment but when running kibana (yarn start --oss), I don't see the embed options as shown in the images above. I would really appreciate it if you could point me in the right direction. Maybe there's a specific branch I need to switch to which contains the changes mentioned here? Thank you in advance.

ajwild commented:
Hi @thabanicus, this hasn't been merged yet, so you'll have to use my branch for now: [ajwild:embed-mode-options](https://github.com/ajwild/kibana/tree/embed-mode-options)

I'm not sure if you'll need to run `yarn kbn bootstrap` before starting the server, probably not, but it might be worth keeping in mind if you're still having issues.

flash1293 commented:
Jenkins, test this.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50556/)
* Commit: 7fd2e18db751af8a52fe159d1e4ccc605814f398
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50556/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50556/execution/node/251/log)

### History
* :green_heart: [Build #49149](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49149/) succeeded 9fb085033c333f670ba06b088a9690a05587d026
* :broken_heart: [Build #49123](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/) failed 6e55e8189b0dc4acd913e4b2b619729747ecb42f
* :broken_heart: [Build #49076](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/) failed 0d92ea5191f7f50e763f12c48c9a763ff6b6debc
* :broken_heart: [Build #47914](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/) failed 07aa92deb835ea89c4063910e4be508e1252be20
* :green_heart: [Build #28714](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/) succeeded bbca5a761e6da5a246f223457264c29e096213e3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/28714/","number":"28714","commit":"bbca5a761e6da5a246f223457264c29e096213e3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/","number":"47914","commit":"07aa92deb835ea89c4063910e4be508e1252be20"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/","number":"49076","commit":"0d92ea5191f7f50e763f12c48c9a763ff6b6debc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/","number":"49123","commit":"6e55e8189b0dc4acd913e4b2b619729747ecb42f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49149/","number":"49149","commit":"9fb085033c333f670ba06b088a9690a05587d026"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50556/","number":"50556","commit":"7fd2e18db751af8a52fe159d1e4ccc605814f398"}]}
PIPELINE-->

lizozom commented:
@elasticmachine merge upstream

ajwild commented:
@lizozom it's not something I require at this moment, and I have to admit I didn't give it too much thought. It seemed like there may be some value in allowing users to view the iframe fullscreen, although I quickly tested this just now and it doesn't seem to work. I also thought that allowing users to share snapshots could be useful, but this would take them into another context without the parent website, so it's probably best to avoid. I see no use for Edit or Clone when embedded. I'd be happy enough to remove it in this PR if @flash1293 agrees.

flash1293 commented:
>  It seemed like there may be some value in allowing users to view the iframe fullscreen, although I quickly tested this just now and it doesn't seem to work

You mean clicking "Fullscreen" in the iframe would make the iframe element in the website embedding it take full width and height? That's not possible without code outside of the iframe which is not something we do right now.

I can see a use case for editing within iframes in some cases (users wouldn't have to jump from the embedding to actual Kibana to quickly change a small thing), so IMHO we could leave it in just for completeness sake. @majagrubic do you think that's fine?

flash1293 commented:
@elassticmachine merge upstream

flash1293 commented:
@elasticmachine merge upstream

flash1293 commented:
Jenkins, test this.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51309/)
* Commit: 0a2a504e93bc3519455801d8a53cc5d765506143
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51309/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51309/execution/node/273/log)

### History
* :broken_heart: [Build #50556](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50556/) failed 7fd2e18db751af8a52fe159d1e4ccc605814f398
* :green_heart: [Build #49149](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49149/) succeeded 9fb085033c333f670ba06b088a9690a05587d026
* :broken_heart: [Build #49123](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/) failed 6e55e8189b0dc4acd913e4b2b619729747ecb42f
* :broken_heart: [Build #49076](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/) failed 0d92ea5191f7f50e763f12c48c9a763ff6b6debc
* :broken_heart: [Build #47914](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/) failed 07aa92deb835ea89c4063910e4be508e1252be20

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/47914/","number":"47914","commit":"07aa92deb835ea89c4063910e4be508e1252be20"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/","number":"49076","commit":"0d92ea5191f7f50e763f12c48c9a763ff6b6debc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/","number":"49123","commit":"6e55e8189b0dc4acd913e4b2b619729747ecb42f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49149/","number":"49149","commit":"9fb085033c333f670ba06b088a9690a05587d026"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50556/","number":"50556","commit":"7fd2e18db751af8a52fe159d1e4ccc605814f398"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51309/","number":"51309","commit":"0a2a504e93bc3519455801d8a53cc5d765506143"}]}
PIPELINE-->

flash1293 commented:
Eslint got updated in the meantime.

Jenkins, test this.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51326/)
* Commit: 7a397621c51b4c12ca6fb07d1ada864fbd21ccbd

### History
* :broken_heart: [Build #51309](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51309/) failed 0a2a504e93bc3519455801d8a53cc5d765506143
* :broken_heart: [Build #50556](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50556/) failed 7fd2e18db751af8a52fe159d1e4ccc605814f398
* :green_heart: [Build #49149](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49149/) succeeded 9fb085033c333f670ba06b088a9690a05587d026
* :broken_heart: [Build #49123](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/) failed 6e55e8189b0dc4acd913e4b2b619729747ecb42f
* :broken_heart: [Build #49076](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/) failed 0d92ea5191f7f50e763f12c48c9a763ff6b6debc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49076/","number":"49076","commit":"0d92ea5191f7f50e763f12c48c9a763ff6b6debc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/","number":"49123","commit":"6e55e8189b0dc4acd913e4b2b619729747ecb42f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49149/","number":"49149","commit":"9fb085033c333f670ba06b088a9690a05587d026"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50556/","number":"50556","commit":"7fd2e18db751af8a52fe159d1e4ccc605814f398"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51309/","number":"51309","commit":"0a2a504e93bc3519455801d8a53cc5d765506143"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51326/","number":"51326","commit":"7a397621c51b4c12ca6fb07d1ada864fbd21ccbd"}]}
PIPELINE-->

majagrubic commented:
> @majagrubic do you think that's fine?

I think it's fine to leave it for now, but I also don't see much need for it. I think it can be removed in a follow-up PR, no need to block this one.

shubham184 commented:
I really want to use this feature, any idea when this will be in production.

flash1293 commented:
@ajwild I just reviewed again and everything works fine expect for the fullscreen mode when embedded - the controls don't hide:

Regular fullscreen mode:
<img width="767" alt="Screenshot 2020-06-04 at 12 05 20" src="https://user-images.githubusercontent.com/1508364/83744000-c70d6500-a65b-11ea-9b5f-36dd39594718.png">

After clicking fullscreen in embedded dashboard with all controls enabled:
<img width="761" alt="Screenshot 2020-06-04 at 12 05 30" src="https://user-images.githubusercontent.com/1508364/83744049-d68cae00-a65b-11ea-970b-89b7ca469aef.png">

Maybe I misunderstood your comment in https://github.com/elastic/kibana/pull/58435#issuecomment-636466173 - when the button is shown I would expect it to hide the controls within the iframe.

Could you either make sure the fullscreen mode works as it does when used in "normal" Kibana or simply hide the fullscreen menu option when embedded? Then we can merge the PR.

ajwild commented:
@flash1293 it wasn't a misunderstanding, it was something I overlooked when refactoring (see https://github.com/elastic/kibana/pull/58435#issuecomment-627010376). It should be working as expected now. I've rebased and fixed a merge conflict too.

flash1293 commented:
@elasticmachine merge upstream

flash1293 commented:
Jenkins, test this

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52315/)
* Commit: 33e8fc4c4d12ff57ad30c7569de5a3c8dd2ea5e6

### History
* :green_heart: [Build #51326](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51326/) succeeded 7a397621c51b4c12ca6fb07d1ada864fbd21ccbd
* :broken_heart: [Build #51309](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51309/) failed 0a2a504e93bc3519455801d8a53cc5d765506143
* :broken_heart: [Build #50556](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50556/) failed 7fd2e18db751af8a52fe159d1e4ccc605814f398
* :green_heart: [Build #49149](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49149/) succeeded 9fb085033c333f670ba06b088a9690a05587d026
* :broken_heart: [Build #49123](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/) failed 6e55e8189b0dc4acd913e4b2b619729747ecb42f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49123/","number":"49123","commit":"6e55e8189b0dc4acd913e4b2b619729747ecb42f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/49149/","number":"49149","commit":"9fb085033c333f670ba06b088a9690a05587d026"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50556/","number":"50556","commit":"7fd2e18db751af8a52fe159d1e4ccc605814f398"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51309/","number":"51309","commit":"0a2a504e93bc3519455801d8a53cc5d765506143"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51326/","number":"51326","commit":"7a397621c51b4c12ca6fb07d1ada864fbd21ccbd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52315/","number":"52315","commit":"33e8fc4c4d12ff57ad30c7569de5a3c8dd2ea5e6"}]}
PIPELINE-->

ajwild commented:
No problem! Thanks to you and everyone else involved for all the helpful feedback and patience.

LeeDr commented:
If you don't include the time filter, what time is used?  Is the current time range from the URL included?  Does the iFrame user know what the time range is?  If it's a relative time range does that still work?

flash1293 commented:
@LeeDr It's using the time range from the url (which can also be relative by using datemath involving `now`).

RonnieDilli commented:
Amazing news guys, these features are deal breakers for us!
Do you have any idea when this might be released?

flash1293 commented:
@RonnieDilli it will be released in the upcoming version 7.9

fbaligand commented:
Great new feature!

periyasamy-sbna commented:
Amazing future. I'm waiting for this release.

pgraca-38 commented:
Hello, pushing that up. Any news on when this will be merged? thanks

flash1293 commented:
@pgraca-38 it's already released with 7.9

pgraca-38 commented:
Thank You! Completly missed it )
